### PR TITLE
LibJS: Reflect a bunch of normative changes in the Temporal spec

### PR DESCRIPTION
**Pull request by linusg**

### CODE REVIEW

**Comment by alimpfard** [comment]
#9710 also does the first change, could we leave it to that PR?

**Comment by linusg** [comment]
@alimpfard sure, dropped the commit :+1:

File: Userland/Libraries/LibJS/Runtime/Temporal/Calendar.cpp [hunk 1]
```diff
@@ -137,12 +137,12 @@ double calendar_year(GlobalObject& global_object, Object& calendar, Object& date
 
     // 3. If result is undefined, throw a RangeError exception.
     if (result.is_undefined()) {
-        vm.throw_exception<RangeError>(global_object, ErrorType::TemporalInvalidCalendarFunctionResult, vm.names.year.as_string());
+        vm.throw_exception<RangeError>(global_object, ErrorType::TemporalInvalidCalendarFunctionResult, vm.names.year.as_string(), "undefined"sv);
```
Comment (IdanHo) [inline Userland/Libraries/LibJS/Runtime/Temporal/Calendar.cpp:140]: how about `vm.names.undefined.as_string()`?

File: Userland/Libraries/LibJS/Runtime/Temporal/Calendar.cpp [hunk 2]
```diff
@@ -137,12 +137,12 @@ double calendar_year(GlobalObject& global_object, Object& calendar, Object& date
...
         return {};
     }
 
-    // 4. Return ? ToIntegerOrInfinity(result).
-    return result.to_integer_or_infinity(global_object);
+    // 4. Return ? ToIntegerThrowOnInfinity(result).
+    return to_integer_throw_on_infinity(global_object, result, ErrorType::TemporalInvalidCalendarFunctionResult, vm.names.year.as_string(), "Infinity"sv);
```
Comment (IdanHo) [inline Userland/Libraries/LibJS/Runtime/Temporal/Calendar.cpp:145]: Same as above

**Review by IdanHo — APPROVED** [review]
Looks good, just a small optional suggestion, nice work on getting your spec fix merged :)

File: Userland/Libraries/LibJS/Runtime/Temporal/Calendar.cpp (hunk 1 above)
Comment (linusg) [reply Userland/Libraries/LibJS/Runtime/Temporal/Calendar.cpp]: Updated, thanks!

**Pull request merged**